### PR TITLE
[Doc] Update inference doc

### PR DESCRIPTION
**Pull request by xiexinch**
## Motivation

as title.

## Modification

- docs/en/user_guides/3_inference.md

### CODE REVIEW

File: docs/en/user_guides/3_inference.md [hunk 1]
```diff
@@ -4,22 +4,130 @@ MMSegmentation provides pre-trained models for semantic segmentation in [Model Z
...
+
+- model (nn.Module) - The loaded segmentor
+- imgs (str, np.ndarray, or list\[str/np.ndarray\]) - Either image files or loaded images
+
+Returns:
+
+- :obj:`SegDataSample` or list\[:obj:`SegDataSample`\]: If imgs is a list or tuple, the same length list type results will be returned, otherwise return the segmentation results directly.
+
+**Note:** [SegDataSample](https://github.com/open-mmlab/mmsegmentation/blob/1.x/mmseg/structures/seg_data_sample.py) is a data structure interface of MMSegmentation, it is used as interfaces between different components. `SegDataSample` implement the abstract data element `mmengine.structures.BaseDataElement`, please refer to [MMEngine](https://mmengine.readthedocs.io/zh_CN/latest/advanced_tutorials/data_element.html) for more information.
+
```
Comment (MeowZheng) [inline docs/en/user_guides/3_inference.md:67]: please refer to data element [documentation](url) in MMEngine

File: docs/en/user_guides/3_inference.md [hunk 2]
```diff
@@ -4,22 +4,130 @@ MMSegmentation provides pre-trained models for semantic segmentation in [Model Z
...
+The attributes in `SegDataSample` are divided into several parts:
+
+- `gt_sem_seg` (PixelData) - Ground truth of semantic segmentation.
+- `pred_sem_seg` (PixelData) - Prediction of semantic segmentation.
+- `seg_logits` (PixelData) - Predicted logits of semantic segmentation.
```
Comment (MeowZheng) [inline docs/en/user_guides/3_inference.md:72]: It would be better to explain what `PixelData` is when it first appeared.

File: docs/en/user_guides/3_inference.md [hunk 3]
```diff
@@ -4,22 +4,130 @@ MMSegmentation provides pre-trained models for semantic segmentation in [Model Z
...
+img_path = 'demo/demo.png'
+
 # register all modules in mmseg into the registries
 register_all_modules()
+
+model = init_model(config_path, checkpoint_path)
+result = inference_model(model, img_path)
+```
+
+### mmseg.apis.show_result_pyplot
```
Comment (MeowZheng) [inline docs/en/user_guides/3_inference.md:91]: what is the function return?

File: docs/en/user_guides/3_inference.md [hunk 4]
```diff
@@ -4,22 +4,130 @@ MMSegmentation provides pre-trained models for semantic segmentation in [Model Z
...
+
+# inference on given image
+result = inference_model(model, img_path)
+
+# visualize the segmentation result
+show_result_pyplot(model, img_path, result)
+
+# save the visualization result
+show_result_pyplot(model, img_path, result, out_file='work_dirs/result.png')
 ```
```
Comment (MeowZheng) [inline docs/en/user_guides/3_inference.md:133]: I think it would be necessary to tell where users can find the saved picture.

File: docs/en/user_guides/3_inference.md [hunk 5]
```diff
@@ -4,22 +4,130 @@ MMSegmentation provides pre-trained models for semantic segmentation in [Model Z
...
-result = inference_model(model, img)
+
+# inference on given image
+result = inference_model(model, img_path)
+
+# visualize the segmentation result
+show_result_pyplot(model, img_path, result)
+
+# save the visualization result
+show_result_pyplot(model, img_path, result, out_file='work_dirs/result.png')
```
Comment (MeowZheng) [inline docs/en/user_guides/3_inference.md:132]: I think we can tell users what is the difference between them, and suggest they use which command in what situation, like if your device doesn't have x11, we suggest saving the visualization results, and if you would like to display the picture in a window, you can use balabala.

File: docs/en/user_guides/3_inference.md [hunk 6]
```diff
@@ -4,22 +4,130 @@ MMSegmentation provides pre-trained models for semantic segmentation in [Model Z
...
 model = init_model(config_file, checkpoint_file, device='cuda:0')
-# test image pair, and save the results
-img = 'demo/demo.png'
-result = inference_model(model, img)
+
+# inference on given image
+result = inference_model(model, img_path)
+
+# visualize the segmentation result
+show_result_pyplot(model, img_path, result)
```
Comment (MeowZheng) [inline docs/en/user_guides/3_inference.md:129]: I think we should add `wait_time` in example, as the default value of it is 0, and users might have question why they don't see anything after run the example.

File: docs/en/user_guides/3_inference.md [hunk 7]
```diff
@@ -4,22 +4,130 @@ MMSegmentation provides pre-trained models for semantic segmentation in [Model Z
...
+checkpoint_path = 'checkpoints/pspnet_r50-d8_512x1024_40k_cityscapes_20200605_003338-2966598c.pth'
+
+
+# register all modules in mmseg into the registries
+register_all_modules()
+
+# init model without checkpoint
+model = init_model(config_path)
+
+# init model with checkpoint
```
Comment (MeowZheng) [inline docs/en/user_guides/3_inference.md:46]: init model and load checkpoint

File: docs/en/user_guides/3_inference.md [hunk 8]
```diff
@@ -4,22 +4,130 @@ MMSegmentation provides pre-trained models for semantic segmentation in [Model Z
...
+model = init_model(config_path, checkpoint_path)
+
+# init model and load model on CPU
```
Comment (MeowZheng) [inline docs/en/user_guides/3_inference.md:49]:
```suggestion
# init model and load checkpoint on CPU
```

File: docs/en/user_guides/3_inference.md [hunk 9]
```diff
@@ -4,22 +4,141 @@ MMSegmentation provides pre-trained models for semantic segmentation in [Model Z
...
+Parameters:
+
+- config (str, :obj:`Path`, or :obj:`mmengine.Config`) - Config file path or the config object.
+- checkpoint (str, optional) - Checkpoint path. If left as None, the model will not load any weights.
+- device (str, optional) - CPU/CUDA device option. Default 'cuda:0'.
+- cfg_options (dict, optional) - Options to override some settings in the used config.
+
+Returns:
+
+- Torch.nn.Module: The constructed segmentor.
```
Comment (MeowZheng) [inline docs/en/user_guides/3_inference.md:28]:
```suggestion
- nn.Module: The constructed segmentor.
```

File: docs/en/user_guides/3_inference.md [hunk 10]
```diff
@@ -4,22 +4,141 @@ MMSegmentation provides pre-trained models for semantic segmentation in [Model Z
...
+from mmseg.utils import register_all_modules
+
+config_path = 'configs/pspnet/pspnet_r50-d8_4xb2-40k_cityscapes-512x1024.py'
+checkpoint_path = 'checkpoints/pspnet_r50-d8_512x1024_40k_cityscapes_20200605_003338-2966598c.pth'
+
+
+# register all modules in mmseg into the registries
+register_all_modules()
+
+# init model without checkpoint
```
Comment (MeowZheng) [inline docs/en/user_guides/3_inference.md:43]: init -> initializate

File: docs/en/user_guides/3_inference.md [hunk 11]
```diff
@@ -4,22 +4,141 @@ MMSegmentation provides pre-trained models for semantic segmentation in [Model Z
...
+Inference image(s) with the segmentor.
+
+Parameters:
+
+- model (nn.Module) - The loaded segmentor
+- imgs (str, np.ndarray, or list\[str/np.ndarray\]) - Either image files or loaded images
+
+Returns:
+
+- :obj:`SegDataSample` or list\[:obj:`SegDataSample`\]: If imgs is a list or tuple, the same length list type results will be returned, otherwise return the segmentation results directly.
```
Comment (MeowZheng) [inline docs/en/user_guides/3_inference.md:64]: obj might be removed

File: docs/en/user_guides/3_inference.md [hunk 12]
```diff
@@ -4,22 +4,141 @@ MMSegmentation provides pre-trained models for semantic segmentation in [Model Z
...
+
+**Note:** [SegDataSample](https://github.com/open-mmlab/mmsegmentation/blob/1.x/mmseg/structures/seg_data_sample.py) is a data structure interface of MMSegmentation, it is used as interfaces between different components. `SegDataSample` implement the abstract data element `mmengine.structures.BaseDataElement`, please refer to data element [documentation](https://mmengine.readthedocs.io/zh_CN/latest/advanced_tutorials/data_element.html) in [MMEngine](https://github.com/open-mmlab/mmengine) for more information.
+
+The attributes in `SegDataSample` are divided into several parts:
+
+- `gt_sem_seg` (PixelData) - Ground truth of semantic segmentation.
+- `pred_sem_seg` (PixelData) - Prediction of semantic segmentation.
+- `seg_logits` (PixelData) - Predicted logits of semantic segmentation.
```
Comment (MeowZheng) [inline docs/en/user_guides/3_inference.md:72]:
```
PixelData -> `PixelData`
```

File: docs/en/user_guides/3_inference.md [hunk 13]
```diff
@@ -4,22 +4,141 @@ MMSegmentation provides pre-trained models for semantic segmentation in [Model Z
...
-# test image pair, and save the results
-img = 'demo/demo.png'
-result = inference_model(model, img)
+model = init_model(config_path, checkpoint_path, device='cuda:0')
+
+# inference on given image
+result = inference_model(model, img_path)
+
+# display the segmentation result
+vis_image = show_result_pyplot(model, img_path, result)
```
Comment (MeowZheng) [inline docs/en/user_guides/3_inference.md:135]: these lines can be removed

File: docs/en/user_guides/3_inference.md [hunk 14]
```diff
@@ -4,22 +4,141 @@ MMSegmentation provides pre-trained models for semantic segmentation in [Model Z
...
+
+Example:
+
+```python
+from mmseg.apis import init_model
+from mmseg.utils import register_all_modules
+
+config_path = 'configs/pspnet/pspnet_r50-d8_4xb2-40k_cityscapes-512x1024.py'
+checkpoint_path = 'checkpoints/pspnet_r50-d8_512x1024_40k_cityscapes_20200605_003338-2966598c.pth'
+
```
Comment (MeowZheng) [inline docs/en/user_guides/3_inference.md:38]:
```suggestion
```

File: docs/en/user_guides/3_inference.md [hunk 15]
```diff
@@ -4,22 +4,140 @@ MMSegmentation provides pre-trained models for semantic segmentation in [Model Z
...
+**Note:** [SegDataSample](https://github.com/open-mmlab/mmsegmentation/blob/1.x/mmseg/structures/seg_data_sample.py) is a data structure interface of MMSegmentation, it is used as interfaces between different components. `SegDataSample` implement the abstract data element `mmengine.structures.BaseDataElement`, please refer to data element [documentation](https://mmengine.readthedocs.io/zh_CN/latest/advanced_tutorials/data_element.html) in [MMEngine](https://github.com/open-mmlab/mmengine) for more information.
+
+The attributes in `SegDataSample` are divided into several parts:
+
+- `gt_sem_seg` (`PixelData`) - Ground truth of semantic segmentation.
+- `pred_sem_seg` (`PixelData`) - Prediction of semantic segmentation.
+- `seg_logits` (`PixelData`) - Predicted logits of semantic segmentation.
+
+**Note** [PixelData](https://github.com/open-mmlab/mmengine/blob/main/mmengine/structures/pixel_data.py) is the data structure for pixel-level annnotations or predictions, please refer to PixelData [documentation](https://mmengine.readthedocs.io/zh_CN/latest/advanced_tutorials/data_element.html#pixeldata) in [MMEngine](https://github.com/open-mmlab/mmengine) for more information.
```
Comment (MeowZheng) [inline docs/en/user_guides/3_inference.md:73]: please provide the docs url of English version.

**Review by MeowZheng — APPROVED** [review]
